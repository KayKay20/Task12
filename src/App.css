body{
  background-image: url('img/blue.jpg');
  background-size: cover;
  background-repeat: no-repeat;
  margin-left: 25%;
  margin-right: 25%;
  margin-top: 10%;
}
.App {
  text-align: center;
}

.Input{
  border-radius: 10px;
  height: 20px;
  width: 250px;
}

.Button{
  color: blue;
}

.col50{
  width: 50%;
  float: left;
  margin-top: 100px;
}

.form {
    margin-top: -20px;
}


.center{
  margin-bottom: -50px;
}

.weather__info {
  margin-top: 10%;
}

.weather__key {
  color:black;
}



