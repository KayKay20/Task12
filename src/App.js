import React from 'react';
import './App.css';
import Form from './components/form.js';
import Background from './img/blue.jpg';
import Weather from './components/Weather.js';
import Titles from './components/title.js';


class App extends React.Component {
  
  state = {
    temperature: undefined,
    city: undefined,
    country: undefined,
    humidity: undefined,
    description: undefined,
    error: undefined
  };

  getWeather = async (e) => {
    e.preventDefault();
    const city = e.target.elements.city.value;
    const country = e.target.elements.country.value;
    const api_call = await fetch(`http://api.openweathermap.org/data/2.5/weather?q=${city},${country}&APPID=4a483a22618f634b50f15e527c11dd11&units=metric`);
    const data = await api_call.json();
    if (city && country) {
    console.log(data);  
    this.setState({
      temperature: data.main.temp,
      city: data.name,
      country: data.sys.country,
      humidity: data.main.humidity,
      description: data.weather[0].description,
      error:""
    });
  } else {
     this.setState({
      temperature: undefined,
      city: undefined,
      country: undefined,
      humidity: undefined,
      description: undefined,
      error:"Please enter the value."
    });


  }
};

  render() {
    return(

  <div className="App">
    <Titles/>
    <br/>
    <Form className="form" getWeather={this.getWeather}/>
    <Weather 
    temperature={this.state.temperature}
    city={this.state.city}
    country={this.state.country}
    humidity={this.state.humidity} 
    description={this.state.description}
    error={this.state.error}
      />
  </div>


      );
  }
}
export default App;
