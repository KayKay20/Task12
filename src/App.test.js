import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import renderer from 'react-test-renderer';

it('renders without crashing', () => {
 const div = document.createElement('div');
 ReactDOM.render(<App />, div);
 ReactDOM.unmountComponentAtNode(div);
});

it('renders correctly', () =>{
    const tree = renderer
    .create(<App />)
    .toJSON();
        expect(tree).toMatchSnapshot()
});

it('the data is peanut butter', async () => {
   const something =  await fetch(`http://api.openweathermap.org/data/2.5/weather?q=Cape Town,ZA&APPID=4a483a22618f634b50f15e527c11dd11&units=metric`);
  const result = await something.json() 
}
);






