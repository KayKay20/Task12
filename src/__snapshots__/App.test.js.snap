// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="App"
>
  <div
    className="center"
  >
    <h1
      className="title"
    >
      Weather Finder
    </h1>
    <p
      className="title"
    >
      Find out temperature, condition and more...
    </p>
  </div>
  <br />
  <section>
    <form
      className="col50"
      onSubmit={[Function]}
    >
      <input
        className="Input"
        name="city"
        placeholder="City..."
        type="text"
      />
      <br />
      <br />
      <input
        className="Input"
        name="country"
        placeholder="Country..."
        type="text"
      />
      <br />
      <button
        className="primary-btn"
      >
        Search
      </button>
    </form>
  </section>
  <div
    className="weather__info"
  />
</div>
`;
